### PR TITLE
Allow creating new object from dict without specifying required fields

### PR DESCRIPTION
**Pull request by reuben**

### CODE REVIEW

**Comment by erogol** [comment]
So the intention is to return a new instance from a dict, right ? 

Also, why do you need `WithRequired()`?

If something is required it should be set `MISSING`. Would that work in your case?

**Comment by reuben** [comment]
`MISSING` effectively makes the field optional again. The validation is too lazy, even calling `check_values` by hand doesn't catch the missing value.

**Comment by reuben** [comment]
> So the intention is to return a new instance from a dict, right ?

Correct. In particular in the case of having fields without default values.

File: coqpit/coqpit.py [hunk 1]
```diff
@@ -389,6 +396,34 @@ def deserialize(self, data: dict) -> "Serializable":
...
+            raise ValueError()
+        data = data.copy()
+        init_kwargs = {}
+        for field in fields(cls):
+            # if field.name == 'dataset_config':
+            if field.name not in data:
+                if field.name in vars(cls):
+                    init_kwargs[field.name] = vars(cls)[field.name]
+                    continue
+                raise ValueError(f'Missing required field "{field.name}"')
```
[the remark this answers is not in the document]
Comment (reuben) [reply coqpit/coqpit.py]: Fixed!

**Comment by erogol** [comment]
`MISSING` causes an error if the field is not set to another value before the field is called.

**Comment by reuben** [comment]
Right, that's too late, I want to to fail on object creation/parsing from the dict, as my objects are invalid without the required fields. Adding MISSING also removes the required parameter from the generated constructor, which I don't want for my classes.

**Comment by erogol** [comment]
Ok I see, we have contrasting requirements. Then it makes sense.

**Review by erogol — APPROVED** [review]

**Pull request merged**